pallets/gear-program/src/migrations/paused_storage.rs

```diff
@@ -0,0 +1,230 @@
+// This file is part of Gear.
+
+// Copyright (C) 2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Config, Pallet};
+use frame_support::{
+    traits::{Get, GetStorageVersion, OnRuntimeUpgrade, StorageVersion},
+    weights::Weight,
+};
+use sp_std::marker::PhantomData;
+
+#[cfg(feature = "try-runtime")]
+use {
+    frame_support::ensure,
+    sp_runtime::{codec::Encode, TryRuntimeError},
+    sp_std::vec::Vec,
+};
+
+pub struct RemovePausedProgramStorageMigration<T: Config>(PhantomData<T>);
+
+const MIGRATE_FROM_VERSION: u16 = 5;
+const MIGRATE_TO_VERSION: u16 = 6;
+const ALLOWED_CURRENT_STORAGE_VERSION: u16 = 6;
```

Comment:
This is not correct versioning, because we have `allocations.rs` migration, which also migrate version to `6`. This migration applies after `allocation` migration, so you should increase pallet version to `7` and set here 
```suggestion
const MIGRATE_FROM_VERSION: u16 = 6;
const MIGRATE_TO_VERSION: u16 = 7;
const ALLOWED_CURRENT_STORAGE_VERSION: u16 = 7;
```

also change ALLOWED_CURRENT_STORAGE_VERSION for other migrations.

Reply:
Currently you migration is skipped